engine/src/main/java/com/sandboni/core/engine/sta/operation/RelatedTestsOperation.java

```diff
@@ -30,7 +32,22 @@
         ShortestPathAlgorithm<Vertex, Edge> algorithm = new BellmanFordShortestPath<>(graph);
         return new SetResult<>(emptyIfFalse(graph.containsVertex(START_VERTEX) && graph.containsVertex(END_VERTEX),
                 () -> allTests.stream()
-                        .filter(v -> isAffectedCucumberVertex(v) || algorithm.getPath(END_VERTEX, v) != null)));
+                        .filter(v -> isAffectedCucumberVertex(v) || algorithm.getPath(END_VERTEX, v) != null)
+                .flatMap(v -> handleSuiteVertex(v, algorithm, graph).stream())
+        ));
+        // todo: handle dups
+    }
+
+    /**
+     * replace any test which is part of a suite with it's suite vertex
+     */
+    private Set<TestVertex> handleSuiteVertex(TestVertex v, ShortestPathAlgorithm<Vertex, Edge> algorithm, Graph<Vertex, Edge> graph) {
+        boolean hasRelatedTestSuite = algorithm.getPath(v, TEST_SUITE_VERTEX) != null;
```

Comment:
I think that at this point, you already have a link between the test and TEST_SUITE_VERTEX, is that correct? If that is the case then you don't need to find a path but only check if TEST_SUITE_VERTEX exists in the list of adjacent nodes to this tests. Please correct me if I'm wrong.

Reply:
TEST_SUITE_VERTEX will not be adjacent (only the TestSuiteVertex), but you are right that I don't need the hasRelatedTestSuite check. I can just filter the adjacent nodes and check for an empty result. will change that.